### PR TITLE
CBL-3861 Add Query / Index Builder CreateIndex API

### PR DESCRIPTION
**Pull request by Sandychuang8**

### CODE REVIEW

File: src/Couchbase.Lite.Tests.Shared/ScopeCollectionTest.cs [hunk 1]
```diff
@@ -820,6 +830,106 @@ public void TestUseScopeAPIAfterDeletingAllCollectionsFromDifferentDBInstance()
...
+            CollA.CreateIndex("myindex", index);
+
+            CollA.GetIndexes().Should().BeEquivalentTo(new[] { "myindex" });
+        }
+
+        [Fact]
+        public void TestCreateSameNameIndexes()
+        {
+            var fName = Expression.Property("firstName");
+            var lName = Expression.Property("lastName");
```
Comment (borrrden) [inline src/Couchbase.Lite.Tests.Shared/ScopeCollectionTest.cs:912]: Unused?

Comment (Sandychuang8) [reply src/Couchbase.Lite.Tests.Shared/ScopeCollectionTest.cs:912]: lName is used in line 918 below..

File: src/Couchbase.Lite.Tests.Shared/ScopeCollectionTest.cs [hunk 2]
```diff
@@ -820,6 +830,106 @@ public void TestUseScopeAPIAfterDeletingAllCollectionsFromDifferentDBInstance()
...
+            CollA.GetIndexes().Count.Should().Be(1);
+            q.Explain().Contains("USING INDEX KeyIndex").Should().BeTrue();
+        }
+
+        [Fact]
+        public void TestCreateIndex()
+        {
+            CollA.GetIndexes().Should().BeEmpty();
+
+            var fName = Expression.Property("firstName");
```
[the remark this answers is not in the document]
Comment (Sandychuang8) [reply src/Couchbase.Lite.Tests.Shared/ScopeCollectionTest.cs:877]: Yes

File: src/Couchbase.Lite.Tests.Shared/ScopeCollectionTest.cs (hunk 1 above)
Comment (borrrden) [reply src/Couchbase.Lite.Tests.Shared/ScopeCollectionTest.cs:910]: oh oops!

**Review by borrrden — APPROVED** [review]
Kind of concerned that there is no information on here addressing the disparity between what the ticket says and what got implemented.  It will go smoother in the future if that is not up in the air.

**Pull request merged**